api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -227,132 +369,209 @@ private void maybeAlertOnCostThresholds(
+    // This is why Math.max is used
+    final Set<DbUser> expiredUsers =
+        dbUsersWithChangedCosts.entrySet().stream()
+            .filter(
+                e ->
+                    costAboveLimit(
+                        e.getValue(),
+                        Math.max(costByCreator.get(e.getKey()), liveCostByCreator.get(e.getKey()))))
+            .map(Entry::getValue)
+            .collect(Collectors.toSet());
 
-      // TODO: prevent setting this limit directly except in this method?
-      user.setFreeTierCreditsLimitDollarsOverride(newDollarLimit);
-      user = userDao.save(user);
+    final Set<DbUser> newlyExpiredFreeTierUsers = Sets.intersection(expiredUsers, freeTierUsers);
 
-      if (userHasRemainingFreeTierCredits(user)) {
-        // may be redundant: enable anyway
-        updateFreeTierWorkspacesStatus(user, BillingStatus.ACTIVE);
-      }
+    logger.info(
+        String.format(
+            "Found %d users exceeding their free tier limit, out of which, %d are new",
+            expiredUsers.size(), newlyExpiredFreeTierUsers.size()));
 
-      userServiceAuditor.fireSetFreeTierDollarLimitOverride(
-          user.getUserId(), previousLimitMaybe, newDollarLimit);
-      return true;
+    for (final DbUser user : newlyExpiredFreeTierUsers) {
+      updateFreeTierWorkspacesStatus(
+          user, BillingStatus.INACTIVE); // Not optimal, should run 1 update statement
     }
+  }
 
-    return false;
+  private boolean costAboveLimit(final DbUser user, final double currentCost) {
+    return CostComparisonUtils.compareCosts(currentCost, getUserFreeTierDollarLimit(user)) > 0;
   }
 
-  /**
-   * Given a workspace, find the amount of free credits that the workspace creator has left.
-   *
-   * @param dbWorkspace The workspace for which to find its creator's free credits remaining
-   * @return The amount of free credits in USD the workspace creator has left, represented as a
-   *     double
-   */
-  public double getWorkspaceCreatorFreeCreditsRemaining(DbWorkspace dbWorkspace) {
-    Double creatorCachedFreeTierUsage = this.getCachedFreeTierUsage(dbWorkspace.getCreator());
-    Double creatorFreeTierDollarLimit = this.getUserFreeTierDollarLimit(dbWorkspace.getCreator());
-    double creatorFreeCreditsRemaining =
-        creatorCachedFreeTierUsage == null
-            ? creatorFreeTierDollarLimit
-            : creatorFreeTierDollarLimit - creatorCachedFreeTierUsage;
-    return Math.max(creatorFreeCreditsRemaining, 0);
+  private Integer getMinutesBeforeLastFreeTierJob() {
+    return Optional.ofNullable(workbenchConfigProvider.get().billing.minutesBeforeLastFreeTierJob)
+        .orElse(120);
+  }
+
+  private Long getNumberOfDaysToConsiderForFreeTierUsageUpdate() {
+    Long numOfDays =
+        workbenchConfigProvider.get().billing.numberOfDaysToConsiderForFreeTierUsageUpdate;
+    if (numOfDays == null) return TWO_DAYS;
+    if (numOfDays > 60) numOfDays = 60l;
+    return numOfDays * 86_400_000;
```

Comment:
See comment above, in general. we use use Duration, or Instant for date calculation. 
But see my comment above, we may dont need this method